.github/workflows/Image-build-AB-diary.yaml

```
@@ -1,119 +1,195 @@
+          echo "Set the branch for current job as $BRANCH"
+
+      - name: checkout
+        uses: actions/checkout@v4
+        if: github.event_name == 'pull_request'
+        with:
+          ref: ${{ env.BRANCH }}
+
+      - name: check diff
+        if: github.event_name == 'pull_request'
+        id: check_diff
         run: |
```

actionlint (1.7.9)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)